library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608TrackRenderer.java

```diff
@@ -334,21 +477,32 @@ private String getDisplayCaption() {
       return null;
     }
 
+    // Every styling span should be closed at end of line
+    closeOpenSpans( STYLE_PRIORITY.STYLE_PRIORITY_ALL );
+
     int endIndex = endsWithNewline ? buildLength - 1 : buildLength;
     if (captionMode != CC_MODE_ROLL_UP) {
-      return captionStringBuilder.substring(0, endIndex);
+      return new SpannableStringBuilder(captionStringBuilder.subSequence(0, endIndex));
```

Comment:
The function signature should be OK with the supertype, but SpannableStringBuilder seems to be necessary to carry the formatting information.

I just found a copy constructor of SpannableStringBuilder for this case:
http://developer.android.com/reference/android/text/SpannableStringBuilder.html#SpannableStringBuilder%28java.lang.CharSequence,%20int,%20int%29